[1.17] Events cleanup

sciwhiz12 opened the pull request:
_This PR is a successor to #7255._

Forge has many events added to its API, to hook into vanilla behaviour and gameplay. These events were implemented by different authors for their usecases, and expanded for general use by other modders. Over the years, as Minecraft and Forge changes their internal structures, these events are updated and modified, or in some cases outright removed if they serve little purpose.

However, there are some flaws in events and their javadocs currently; here are the most notable:

1. Some, if not most events have insufficient documentation on their functions, usage, and event firing information.
1. The method names across some events are not consistent: one event may have `#getRenderPartialTick()`, another has `#getPartialTicks()`, `#getPartialRenderTicks()`, etc.
1. Some events do not encapsulate the fields, so field renaming (which is, in my opinion, an implementation and non-public facing operation) can break accessors.
1. _NEW!_ With the introduction of Forge 1.17, class names have now moved from MCP names to Mojang names. This means that javadocs and event class/method/field names now need to be renamed and fix to be in line with the new class names.

This PR aims to address these points, by a) improving and making consistent the javadocs on all event classes, b) renaming classes, methods, and fields to be in line with the new class names, and c) performing necessary cleanup of events (such as package moving, refactors, and others).

> **Note:** As of writing, this is a draft PR. I plan to move across a lot of my changes from my old 1.16 event cleanup PR to this new one for 1.17.

sciwhiz12 committed:
Cleanup formatting for events

sciwhiz12 committed:
Refactor and rename in client events package

sciwhiz12 committed:
Refactor and rename main events package

sciwhiz12 committed:
Cleanup forge event factory and hooks

sciwhiz12 committed:
Fix compilation errors in tests

sciwhiz12 committed:
Rename `world` events subpackage to `level`

sciwhiz12 committed:
Generate patches

josephtarbit commented:
I'm not sure whether this is note worthy enough but there's discrepancies in subclass/nested event naming - some have an Event suffix but others have no suffix. (E.g ScreenEvent.InitEvent and then TickEvent.Pre)

sciwhiz12 commented:
From what can be seen in the current event naming, at least the `Pre` and `Post` event subclass names (and any specialized subclass extending them) are special cases, and shouldn't be suffixed with `Event`.

sciwhiz12 committed:
Merge latest 1.17.x changes

sciwhiz12 committed:
Add package-infos with nonnull annotations

sciwhiz12 committed:
Move rendering events to `render` subpackage

sciwhiz12 committed:
Remove `furnace`, `brewing`, and `enchanting` subpackages

There are too few events within these subpackages to warrant their
existence.

sciwhiz12 committed:
Rename FOV-related events to better fit their use

sciwhiz12 committed:
Extend `PotionApplicableEvent` to wither boss and skeleton

Since this is used for spiders and poison, it stands to reason that
the event should also be used for wither boss/skeletons and wither.

sciwhiz12 committed:
Cleanup nullability annotations

Also does some small miscellaneous cleanups leftover from previous commits,
such as a misspelled parameter and missing spaces between constructors
and fields.

Some constructors are modified to directly assign values to fields instead
of using the setter, because IntelliJ complains about the field not being
set.

sciwhiz12 committed:
Remove deprecated `UseHoeEvent

thedarkcolour commented:
How about adding a note or annotation in the docs specifying what Dist the event is fired on?

sciwhiz12 commented:
@thedarkcolour: That is part of the event javadocs styling that I iterated on while I did initial (client events) javadocs work on 1.16 in [a branch on my fork](https://github.com/sciwhiz12/MinecraftForge/tree/48f5bb4245db7290c9d0897695de65149fbe8a7c/src/main/java/net/minecraftforge/client/event).

Here's an example javadoc from that branch, for the [`ClientChatEvent`](https://github.com/sciwhiz12/MinecraftForge/blob/48f5bb4245db7290c9d0897695de65149fbe8a7c/src/main/java/net/minecraftforge/client/event/ClientChatEvent.java#L29-L42):
```java
/**
 * <p>Fired when the client is about to send a chat message or command to the server. <br/>
 * This can be used to implement client-only commands as chat messages, or intercepting messages for typos. </p>
 *
 * <p>This event is {@linkplain Cancelable cancelable}, and does not {@linkplain HasResult have a result}. <br/>
 * If the event is cancelled, the chat message or command will not be sent to the server. </p>
 *
 * <p>This event is fired on the {@linkplain MinecraftForge#EVENT_BUS main Forge event bus},
 * only on the {@linkplain LogicalSide#CLIENT logical client}. </p>
 *
 * @see ForgeEventFactory#onClientSendMessage(String)
 **/
@Cancelable
public class ClientChatEvent extends Event
```

Here's another example, for [`ColorHandlerEvent` and its subclass, `ColorHandlerEvent.Block`](https://github.com/sciwhiz12/MinecraftForge/blob/48f5bb4245db7290c9d0897695de65149fbe8a7c/src/main/java/net/minecraftforge/client/event/ColorHandlerEvent.java#L34-L53):
```java
/**
 * <p>Fired for registering block and item color handlers at the appropriate time. <br/>
 * See the two subclasses for registering blocks or items color handlers. </p>
 *
 * @see ColorHandlerEvent.Block
 * @see ColorHandlerEvent.Item
 **/
public abstract class ColorHandlerEvent extends Event implements IModBusEvent
{
    /**
     * <p>Fired for registering block colors handlers. </p>
     *
     * <p>This event is not {@linkplain Cancelable cancelable}, and do not {@linkplain HasResult have a result}. </p>
     *
     * <p>This event is fired on the {@linkplain FMLJavaModLoadingContext#getModEventBus() mod-specific event bus},
     * only on the {@linkplain LogicalSide#CLIENT logical client}. </p>
     *
     * @see ForgeHooksClient#onBlockColorsInit(BlockColors)
     */
    public static class Block extends ColorHandlerEvent
```

sciwhiz12 committed:
Update to latest 1.17.x

sciwhiz12 committed:
Copy javadocs from my 1.16 docs branch for client events

Only the events under `net.minecraftforge.client.event` and
`net.minecraftforge.client.event.render` that existed sometime during
1.16 are currently documented. Javadocs for the remaining client events
are a work-in-progress.

sciwhiz12 committed:
Rename to ParticleProviderRegisterEvent

sciwhiz12 committed:
Add docs for EntityRenderersEvent

sciwhiz12 committed:
Add docs for ClientPlayerChangeGameTypeEvent

gigaherz commented on src/main/java/net/minecraftforge/client/event/EntityRenderersEvent.java:
```diff
@@ -63,26 +89,33 @@ public void registerLayerDefinition(ModelLayerLocation layerLocation, Supplier<L
     {
         /**
-         * Registers an entity renderer.
+         * Registers an entity renderer for the given block entity type.
```
-block

gigaherz commented on src/main/java/net/minecraftforge/client/event/EntityRenderersEvent.java:
```diff
@@ -63,26 +89,33 @@ public void registerLayerDefinition(ModelLayerLocation layerLocation, Supplier<L
-         * @param blockEntityType The BlockEntityType to register a renderer for
-         * @param blockEntityRendererProvider The renderer provider, can be a lambda like MyRenderer::new
+         * @param blockEntityType The block entity type to register a renderer for
+         * @param blockEntityRendererProvider The renderer provider
```
Small inconsistency: you uncapitalized other params, but not these.

gigaherz reviewed (approved):
A couple nitpicks.
LGTM with those little changes

EDIT: Disregard this approval, since I didn't realize I was reviewing one commit in a bigger PR.

sciwhiz12 committed:
Add docs for RegisterClientReloadListenersEvent

sciwhiz12 committed:
Respond to code review by @gigaherz, fix some spacing

sciwhiz12 committed:
Fix non-inlined return tag

sciwhiz12 committed:
Add docs for RegisterShadersEvent

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/ColorHandlerEvent.java:
```diff
@@ -19,17 +19,40 @@
- * Use these events to register block/item
- * color handlers at the appropriate time.
+ * Fired for registering block and item color handlers at the appropriate time.
+ * See the two subclasses for registering blocks or items color handlers.
```
This should be "block or item color handlers" (without the `s` after 'block' and 'item') as on the line above it

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/ColorHandlerEvent.java:
```diff
@@ -19,17 +19,40 @@
 public abstract class ColorHandlerEvent extends Event implements IModBusEvent
 {
+    /**
+     * Fired for registering block colors handlers.
```
"block color handlers", no 's' after 'color'

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/EntityRenderersEvent.java:
```diff
@@ -23,38 +23,64 @@
 import java.util.function.Supplier;
 
+/**
+ * Fired for on different events/actions relating to {@linkplain EntityRenderer entity renderers}.
```
The word 'on' should be left out here

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/ScreenOpenEvent.java:
```diff
@@ -0,0 +1,66 @@
+import net.minecraftforge.fml.LogicalSide;
+
+/**
+ * Fired before any {@link Screen} will be opened, to allow changing or preventing it from being opened.
```
"to allow changing `it` or preventing it from being opened" would make the sentence a bit more clear I think

sciwhiz12 replied:
I feel like the current wording is clear enough, as `changing` in the sentence has no other possible subject than `it` (which would refer to the screen).

SuperMartijn642 replied:
It's more that the way I read it now, is "to allow changing it from being opened or preventing it from being opened"

sciwhiz12 replied:
Fair point.

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java:
```diff
@@ -20,56 +20,88 @@
 
-
+/**
+ * Fired before and after a texture atlas stitched together.
```
I think this should be "after a texture atlas `is` stitched together"

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/render/EntityViewRenderEvent.java:
```diff
@@ -0,0 +1,390 @@
+    }
+
+    /**
+     * Fired for altering the raw field of vision (FOV).
```
Shouldn't this be 'field of view'? The event is also called FieldOfView and 'field of vision' sounds a bit odd, although that might just be me.

sciwhiz12 replied:
I think I might have mixed the word `vision` and `view` there. It should be `field of view`.

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/render/RenderBlockOverlayEvent.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    /**
+     * {@return the block from which the overlay is gotten from}
```
I think one 'from' should be left out here

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/render/RenderBlockOverlayEvent.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    /**
+     * {@return the position of the block from which the overlay is gotten from}
```
Also here, one 'from' should be left out

SuperMartijn642 commented on src/main/java/net/minecraftforge/client/event/render/RenderLivingEvent.java:
```diff
@@ -0,0 +1,155 @@
+
+    /**
+     * {@return the partial tick}
+     * TODO: rename to getPartialTick()
```
Wouldn't this be the time to rename the method?

sciwhiz12 replied:
Yes, but I added that because I am trying to avoid mixing adding documentation and changing any class members in one commit. I planned to (and will) rename the method as the TODO notes.

SuperMartijn642 replied:
Ah ok, that makes sense

SuperMartijn642 reviewed (commented):
Just some remarks on some the javadocs

Harleyoc1 commented:
Reminder that the removal of `EntityEvent$CanUpdate` as was included in the superseded PR could/should be included here in order to close #6394.

sciwhiz12 committed:
Respond to code review by @SuperMartijn642

sciwhiz12 committed:
Rename method in RenderLivingEvent as marked

sciwhiz12 committed:
Update to latest 1.17.x

sciwhiz12 committed:
Fix broken javadocs in RenderLivingEvent

sciwhiz12 committed:
Correct minor spelling mistake
